daemon/api_sideload_n_try.go

```diff
@@ -120,7 +120,8 @@ func (f *Form) GetSnapFiles() ([]*uploadedSnap, *apiError) {
 
 type sideloadFlags struct {
 	snapstate.Flags
-	dangerousOK bool
+	dangerousOK   bool
+	transactional bool
```

Comment:
The problem I see with that is that afaiu `snapstate.Flags` contains options that are per-snap and most of them apply to the full life cycle of the snap, while `transactional` is for a set of snaps and is used only on installation (there is no need to keep track of the flag after that). Also, not sure how to fit that when installing from the store, as `InstallMany()` and the others create flags later than when sideloading.

But I agree that a boolean does not say much when you see it in the method call. Maybe an option would be to include another flag in `daemon.snapInstruction`, but that one is not used when sideloading. Another option would be to create a type for the flag, like:

```
type Transactional int

const (
        TrasactionalTrue Transactional = iota
        TransactionalFalse
)
```
Let me know your thoughts on this.

Reply:
I agree about not passing bool, we try to avoid this, in particular around snapstate functions which often get new flags over time. But @alfonsosanchezbeato is right, `snapstate.Flags` may not be a good choice because it is serialized with `SnapState`. But I'm not a fan of passing Transactional as an extra arg; can make it part of sideloadFlags and just pass sideloadFlags down to InstallMany, or is it problematic?

Reply:
I see, that's a good point. It sounds like we want to pass two types of flags into these functions, persistent/serializable and ephemeral. Maybe we can have a new flags type that contains those two types and then use them accordingly in the functions. It might be overkill now but if we add more flags over time, like Pawel pointed out, then it might be useful. If we don't want to do that, we can at least avoid passing the "raw" boolean into the function and instead assign it to a variable so it documents what it is, like:
```
inTransaction := true
InstallMany(..., inTransaction)
```
wdyt?

Reply:
I see that there is already a `RemoveFlags` type (with only one flag), so maybe it would make sense to add `InstallFlags` and `UpdateFlags` - or just a type named `ActionFlags` that would apply to both, any of these with a `Transaction` bool. But, the problem with abstracting before there is a need is that it is difficult to know how the system will evolve. We don't know if a new flag would apply only to installs, or only to removals, or to both, or if it will be something different that will need more re-thinking, or if the new flag thing will really happen.

So, I'm more inclined to just use transaction for the moment, I'm happy to do it the way you suggest if you prefer that to adding a transactional enum.

Reply:
I think it's fine. The enum feels a bit clunky since there's only two possible values, which conceptually map to true or false, but no approach is without its issues, so I'm ok with it. Thanks!